ts/features/lollipop/saga/index.ts

```diff
@@ -61,6 +73,37 @@ export function* deleteCurrentLollipopKeyAndGenerateNewKeyTag() {
   yield* put(lollipopKeyTagSave({ keyTag: newKeyTag }));
 }
 
+export function* lollipopKeyCheckWithServer() {
+  const maybeSessionInformation: ReturnType<typeof sessionInfoSelector> =
+    yield* select(sessionInfoSelector);
+
+  const publicKey: ReturnType<typeof lollipopPublicKeySelector> = yield* select(
+    lollipopPublicKeySelector
+  );
+
+  if (O.isSome(maybeSessionInformation)) {
```

Comment:
What do you think about rewriting this function body as a functional pipe? Something like
```
  const lollipopAssertionCheckSucceeded = pipe(
    maybeSessionInformation,
    O.chainNullableK(sessionInformation => sessionInformation.lollipopAssertionRef),
    O.chain(sessionLollipopAssertionRef => 
      pipe(
        maybePublicKey,
        O.map(publicKey => {
          const publicKeyThumbprint = jwkThumbprintByEncoding(publicKey, DEFAULT_LOLLIPOP_HASH_ALGORITHM_CLIENT, "base64url");
          return `${DEFAULT_LOLLIPOP_HASH_ALGORITHM_SERVER}-${publicKeyThumbprint}`;
        }),
        O.map(localLollipopAssertionRef => localLollipopAssertionRef === sessionLollipopAssertionRef)
      )
    ),
    O.getOrElse(() => false)
  );

  if (!lollipopAssertionCheckSucceeded) {
    yield* put(sessionInvalid());
    yield* call(restartCleanApplication);
  }
```